curl: update to 8.19.0

herbygillot opened the pull request:
###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? <!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [ ] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

ryandesign commented:
There are a couple considerations needed with this update beyond just updating the version.

In 8.14.0, wcurl is bundled. In 8.15.0, the Secure Transport backend was removed.

ryandesign commented:
The purpose of the developer bundling wcurl with curl is so that wcurl would always be available when curl is installed. I planned to obsolete the wcurl port, replacing it with curl and using the [deactivate hack](https://trac.macports.org/wiki/PortfileRecipes#deactivatehack) in the curl port. I assumed curl's `make install` would install it but I haven't checked.

ryandesign commented:
If we update to 8.14.1 now and tackle the wcurl move, then we can deal with 8.15.0 and the Secure Transport removal later.

herbygillot commented:
@ryandesign, these commits attempt to handle all the issues you've listed at once.

ryandesign commented:
Or just add the patches to the existing libressl port. This presumes the patches can be located and are usable. That can be a ticket to be dealt with later.

The user experience of trying to use the libressl port is not great because it conflicts with the openssl port. There are tickets. 

I was proposing that the darwinssl variant could use Apple libressl.

ryandesign commented:
See https://github.com/jeroen/apple-libressl-sdk

ryandesign commented:
Which I got from this thread where the topic is being discussed again:

https://curl.se/mail/lib-2025-08/0046.html

It sounds like the wolfssl option already supports certificates in the keychain and it's desired to add support for that to the other ssl options later.

CL-Jeremy commented:
Hi,

Following PR #29376, I've prepared https://github.com/CL-Jeremy/macports-ports/commit/1d375fabb9b40ebc43503d7727cbf88d82e84c21 to extend the backend choice for HTTP3 support, as cURL currently requires using the same backend for TCP/QUIC, and, since the majority of ngtcp2 backends could coexist with each other, we could keep features like system keychain integration without sacrificing HTTP3 support. The idea is to match the variant choice of ports bidirectionally (cf. `py-cairo`), so not only will the matching variant is pulled in when the user specifies variant in `curl`, but the (most probably intended) variant will be chosen if a custom variant of `ngtcp2` is already installed. The user could still specify alternative backends, as long as it is available – e. g. having `ngtcp2 +gnutls+openssl+wolfssl`, though `curl +wolfssl` is chosen by default, the user could override this with `curl +ssl` (brand new upstream support with vanilla OpenSSL >= 3.5.0) or `curl +gnutls` (the stable choice before this PR).

Hopefully this could shed some light on how the `curl` port moves forward (after the case of `wcurl` is settled, I personally don't see much problem with the current proposal). If `libressl` or some variant of it (patched or bundled in OS) is to be supported as well, I could add that to `ngtcp2` as well.

reneeotten commented:
@herbygillot @ryandesign looks like this PR is ready to merge? Can we please make progress on this?

mohd-akram commented:
What's the update on this?

It's not clear what the `darwinssl` variant achieves right now, it should probably be removed. In its place, it seems like it would make sense to create a `sectrust` variant and to check if it's enabled inside the `ssl` and `gnutls` variants, and if so, append `--with-apple-sectrust` to `configure.args`. This variant will let curl use the [system certificate store](https://curl.se/docs/sslcerts.html) by default which achieves the same thing as Secure Transport. Therefore, as @CL-Jeremy mentioned, there would be no need to patch libressl or delay this update.

@ryandesign Thoughts on this?

herbygillot commented:
There is another PR for updating curl now ( https://github.com/macports/macports-ports/pull/30739 ), but I've just updated this to version 8.18.0 now, and added a SecTrust variant.

herbygillot committed:
curl: update to 8.19.0

- wcurl is now part of curl
- remove Secure Transport support
- add variant for Apple SecTrust
- mbedtls variant now requires mbedtls3
- update curl-ca-bundle to 20260212
- add patch reverting commit f07a98a which was breaking Clang compilation

herbygillot commented:
> Updated to 8.19.0, but the CI is failing because the Mozilla `certdata` server location appears to having issues...?

OK, this is fixed.

barracuda156 commented:
Looks like distfile now is different for `curl-ca-bundle`:
```
set certdata_distfile       mozilla-central-${certdata_commit}${certdata_extract_suffix}
```

barracuda156 commented:
@herbygillot I think this should be addressed: 
```
:info:build libtool: compile:  /usr/bin/gcc-4.2 -std=gnu99 -DHAVE_CONFIG_H -I../include -I../lib -I../lib -I. -DBUILDING_LIBCURL -DCURL_HIDDEN_SYMBOLS -isystem /opt/bootstrap/include -isystem /opt/bootstrap/include -isystem /opt/bootstrap/libexec/openssl3/include -fvisibility=hidden -pipe -Os -mmacosx-version-min=10.6 -arch x86_64 -Werror-implicit-function-declaration -Wno-system-headers -Wno-shadow -Wno-unreachable-code -Wno-overlength-strings -Wno-missing-field-initializers -MT libcurl_la-mime.lo -MD -MP -MF .deps/libcurl_la-mime.Tpo -c mime.c  -fno-common -DPIC -o .libs/libcurl_la-mime.o
:info:build mv -f .deps/libcurl_la-md5.Tpo .deps/libcurl_la-md5.Plo
:info:build libtool: compile:  /usr/bin/gcc-4.2 -std=gnu99 -DHAVE_CONFIG_H -I../include -I../lib -I../lib -I. -DBUILDING_LIBCURL -DCURL_HIDDEN_SYMBOLS -isystem /opt/bootstrap/include -isystem /opt/bootstrap/include -isystem /opt/bootstrap/libexec/openssl3/include -fvisibility=hidden -pipe -Os -mmacosx-version-min=10.6 -arch x86_64 -Werror-implicit-function-declaration -Wno-system-headers -Wno-shadow -Wno-unreachable-code -Wno-overlength-strings -Wno-missing-field-initializers -MT libcurl_la-mprintf.lo -MD -MP -MF .deps/libcurl_la-mprintf.Tpo -c mprintf.c  -fno-common -DPIC -o .libs/libcurl_la-mprintf.o
:info:build /bin/sh ../libtool  --tag=CC   --mode=compile /usr/bin/gcc-4.2 -std=gnu99 -DHAVE_CONFIG_H   -I../include -I../lib -I../lib -I.  -DBUILDING_LIBCURL  -DCURL_HIDDEN_SYMBOLS -isystem /opt/bootstrap/include -isystem /opt/bootstrap/include -isystem /opt/bootstrap/libexec/openssl3/include -fvisibility=hidden -pipe -Os -mmacosx-version-min=10.6 -arch x86_64 -Werror-implicit-function-declaration -Wno-system-headers -Wno-shadow -Wno-unreachable-code -Wno-overlength-strings -Wno-missing-field-initializers  -MT libcurl_la-multi_ev.lo -MD -MP -MF .deps/libcurl_la-multi_ev.Tpo -c -o libcurl_la-multi_ev.lo `test -f 'multi_ev.c' || echo './'`multi_ev.c
:info:build mprintf.c: In function 'out_double':
:info:build mprintf.c:676: error: #pragma GCC diagnostic not allowed inside functions
:info:build mprintf.c:677: error: #pragma GCC diagnostic not allowed inside functions
:info:build mprintf.c:691: error: #pragma GCC diagnostic not allowed inside functions
:info:build make[2]: *** [libcurl_la-mprintf.lo] Error 1
:info:build make[2]: *** Waiting for unfinished jobs....
:info:build libtool: compile:  /usr/bin/gcc-4.2 -std=gnu99 -DHAVE_CONFIG_H -I../include -I../lib -I../lib -I. -DBUILDING_LIBCURL -DCURL_HIDDEN_SYMBOLS -isystem /opt/bootstrap/include -isystem /opt/bootstrap/include -isystem /opt/bootstrap/libexec/openssl3/include -fvisibility=hidden -pipe -Os -mmacosx-version-min=10.6 -arch x86_64 -Werror-implicit-function-declaration -Wno-system-headers -Wno-shadow -Wno-unreachable-code -Wno-overlength-strings -Wno-missing-field-initializers -MT libcurl_la-mqtt.lo -MD -MP -MF .deps/libcurl_la-mqtt.Tpo -c mqtt.c  -fno-common -DPIC -o .libs/libcurl_la-mqtt.o
:info:build libtool: compile:  /usr/bin/gcc-4.2 -std=gnu99 -DHAVE_CONFIG_H -I../include -I../lib -I../lib -I. -DBUILDING_LIBCURL -DCURL_HIDDEN_SYMBOLS -isystem /opt/bootstrap/include -isystem /opt/bootstrap/include -isystem /opt/bootstrap/libexec/openssl3/include -fvisibility=hidden -pipe -Os -mmacosx-version-min=10.6 -arch x86_64 -Werror-implicit-function-declaration -Wno-system-headers -Wno-shadow -Wno-unreachable-code -Wno-overlength-strings -Wno-missing-field-initializers -MT libcurl_la-multi.lo -MD -MP -MF .deps/libcurl_la-multi.Tpo -c multi.c  -fno-common -DPIC -o .libs/libcurl_la-multi.o
:info:build libtool: compile:  /usr/bin/gcc-4.2 -std=gnu99 -DHAVE_CONFIG_H -I../include -I../lib -I../lib -I. -DBUILDING_LIBCURL -DCURL_HIDDEN_SYMBOLS -isystem /opt/bootstrap/include -isystem /opt/bootstrap/include -isystem /opt/bootstrap/libexec/openssl3/include -fvisibility=hidden -pipe -Os -mmacosx-version-min=10.6 -arch x86_64 -Werror-implicit-function-declaration -Wno-system-headers -Wno-shadow -Wno-unreachable-code -Wno-overlength-strings -Wno-missing-field-initializers -MT libcurl_la-multi_ev.lo -MD -MP -MF .deps/libcurl_la-multi_ev.Tpo -c multi_ev.c  -fno-common -DPIC -o .libs/libcurl_la-multi_ev.o
:info:build mv -f .deps/libcurl_la-imap.Tpo .deps/libcurl_la-imap.Plo
:info:build libtool: compile:  /usr/bin/gcc-4.2 -std=gnu99 -DHAVE_CONFIG_H -I../include -I../lib -I../lib -I. -DBUILDING_LIBCURL -DCURL_HIDDEN_SYMBOLS -isystem /opt/bootstrap/include -isystem /opt/bootstrap/include -isystem /opt/bootstrap/libexec/openssl3/include -fvisibility=hidden -pipe -Os -mmacosx-version-min=10.6 -arch x86_64 -Werror-implicit-function-declaration -Wno-system-headers -Wno-shadow -Wno-unreachable-code -Wno-overlength-strings -Wno-missing-field-initializers -MT libcurl_la-multi_ev.lo -MD -MP -MF .deps/libcurl_la-multi_ev.Tpo -c multi_ev.c -o libcurl_la-multi_ev.o >/dev/null 2>&1
:info:build libtool: compile:  /usr/bin/gcc-4.2 -std=gnu99 -DHAVE_CONFIG_H -I../include -I../lib -I../lib -I. -DBUILDING_LIBCURL -DCURL_HIDDEN_SYMBOLS -isystem /opt/bootstrap/include -isystem /opt/bootstrap/include -isystem /opt/bootstrap/libexec/openssl3/include -fvisibility=hidden -pipe -Os -mmacosx-version-min=10.6 -arch x86_64 -Werror-implicit-function-declaration -Wno-system-headers -Wno-shadow -Wno-unreachable-code -Wno-overlength-strings -Wno-missing-field-initializers -MT libcurl_la-mqtt.lo -MD -MP -MF .deps/libcurl_la-mqtt.Tpo -c mqtt.c -o libcurl_la-mqtt.o >/dev/null 2>&1
:info:build libtool: compile:  /usr/bin/gcc-4.2 -std=gnu99 -DHAVE_CONFIG_H -I../include -I../lib -I../lib -I. -DBUILDING_LIBCURL -DCURL_HIDDEN_SYMBOLS -isystem /opt/bootstrap/include -isystem /opt/bootstrap/include -isystem /opt/bootstrap/libexec/openssl3/include -fvisibility=hidden -pipe -Os -mmacosx-version-min=10.6 -arch x86_64 -Werror-implicit-function-declaration -Wno-system-headers -Wno-shadow -Wno-unreachable-code -Wno-overlength-strings -Wno-missing-field-initializers -MT libcurl_la-mime.lo -MD -MP -MF .deps/libcurl_la-mime.Tpo -c mime.c -o libcurl_la-mime.o >/dev/null 2>&1
:info:build mv -f .deps/libcurl_la-multi_ev.Tpo .deps/libcurl_la-multi_ev.Plo
:info:build mv -f .deps/libcurl_la-mqtt.Tpo .deps/libcurl_la-mqtt.Plo
:info:build mv -f .deps/libcurl_la-mime.Tpo .deps/libcurl_la-mime.Plo
:info:build libtool: compile:  /usr/bin/gcc-4.2 -std=gnu99 -DHAVE_CONFIG_H -I../include -I../lib -I../lib -I. -DBUILDING_LIBCURL -DCURL_HIDDEN_SYMBOLS -isystem /opt/bootstrap/include -isystem /opt/bootstrap/include -isystem /opt/bootstrap/libexec/openssl3/include -fvisibility=hidden -pipe -Os -mmacosx-version-min=10.6 -arch x86_64 -Werror-implicit-function-declaration -Wno-system-headers -Wno-shadow -Wno-unreachable-code -Wno-overlength-strings -Wno-missing-field-initializers -MT libcurl_la-multi.lo -MD -MP -MF .deps/libcurl_la-multi.Tpo -c multi.c -o libcurl_la-multi.o >/dev/null 2>&1
:info:build mv -f .deps/libcurl_la-multi.Tpo .deps/libcurl_la-multi.Plo
:info:build make[2]: Leaving directory `/opt/bootstrap/var/macports/build/curl-f4948e5b/work/curl-8.19.0-x86_64/lib'
:info:build make[1]: *** [all] Error 2
:info:build make[1]: Leaving directory `/opt/bootstrap/var/macports/build/curl-f4948e5b/work/curl-8.19.0-x86_64/lib'
:info:build make: *** [all-recursive] Error 1
:info:build make: Leaving directory `/opt/bootstrap/var/macports/build/curl-f4948e5b/work/curl-8.19.0-x86_64'
:info:build Command failed:  cd "/opt/bootstrap/var/macports/build/curl-f4948e5b/work/curl-8.19.0-x86_64" && /usr/bin/make -j6 -w all 
:info:build Exit code: 2
```
We do have modern compilers, of course, but if `curl` gonna require a modern gcc/clang, bootstrap time for curl gonna skyrocket. (And without modern `curl` the base does not function properly.)

herbygillot committed:
wcurl: replaced by curl

herbygillot committed:
p5-www-curl: patch for CURL_STRICTER removal in curl 8.15

- bump revision for curl 8.19.0 update

herbygillot committed:
privoxy-pki-bundle: bump revision for curl update
